### PR TITLE
Implement experimental support for marimo

### PR DESCRIPTION
**Pull request by bkneis**
Add experimental support for marimo.

Tested using `./devpod-cli --ide marimo up https://github.com/microsoft/vscode-remote-try-python`

### CODE REVIEW

File: cmd/up.go [hunk 1]
```diff
@@ -63,6 +66,7 @@ type UpCmd struct {
 
 	DotfilesSource string
 	DotfilesScript string
+	Token          string
```
Comment (pascalbreuninger) [inline cmd/up.go:69]: What do you think of putting this token into marimo specific settings? Similar to the `bindaddressoption`. 
This way we don't have to pollute all commands with it and can read the ide configuration on the agent through `workspaceInfo.IDE`

Comment (bkneis) [reply cmd/up.go:69]: 100%

Comment (bkneis) [reply cmd/up.go:69]: actually I can't do that unless we are happy with the access token not changing? Currently we generate a new token every connection and exchange over a secure connection. Using the settings would require that the token not change :thinking:

File: pkg/ide/marimo/marimo.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	out, err := exec.Command(args[0], args[1:]...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error installing jupyter notebook: %w", command.WrapCommandError(out, err))
+	}
+	return s.start()
+}
+
+func (s *Server) start() error {
+	return single.Single("marimo.pid", func() (*exec.Cmd, error) {
+		s.log.Infof("Starting jupyter notebook in background...")
```
Comment (pascalbreuninger) [inline pkg/ide/marimo/marimo.go]: 🤭

File: pkg/ide/marimo/marimo.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		args = append(args, "su", s.userName, "-c", runCommand)
+	} else {
+		args = append(args, "sh", "-c", runCommand)
+	}
+
+	// install
+	s.log.Infof("Installing marimo...")
+	out, err := exec.Command(args[0], args[1:]...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error installing jupyter notebook: %w", command.WrapCommandError(out, err))
```
Comment (pascalbreuninger) [inline pkg/ide/marimo/marimo.go]: here as well

File: pkg/ide/ideparse/parse.go [hunk 4]
```diff
@@ -143,6 +143,13 @@ var AllowedIDEs = []AllowedIDE{
 		Icon:         "https://devpod.sh/assets/positron.svg",
 		Experimental: true,
 	},
+	{
+		Name:         config.IDEMarimo,
+		DisplayName:  "Marimo",
+		Options:      vscode.Options,
+		Icon:         "https://devpod.sh/assets/positron.svg",
```
Comment (pascalbreuninger) [inline pkg/ide/ideparse/parse.go]: need an icon

Comment (pascalbreuninger) [reply pkg/ide/ideparse/parse.go]: nevermind :D

File: cmd/up.go (hunk 1 above)
Comment (pascalbreuninger) [reply cmd/up.go:69]: Is that a problem for any given session? 
You could stop it, change the token and restart the workspace 🤷‍♂️

Comment (bkneis) [reply cmd/up.go]: True, updated :)

**Review by pascalbreuninger — APPROVED** [review]
LGTM

**Review by pascalbreuninger — APPROVED** [review]
LGTM

**Pull request merged**